File: raptorsRobotsApp/src/main/java/pl/raptors/raptorsRobotsApp/domain/movement/MovementPath.java
package pl.raptors.raptorsRobotsApp.domain.movement;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@NoArgsConstructor
@Data
@Document(collection = "movement_paths")
public class MovementPath {

    @Id
    private String id;
    private String name;
    private List<UniversalPoint> points;
    private String startStandId;
    private String finishStandId;

    public MovementPath(String name, List<UniversalPoint> points, String startStandId, String finishStandId) {
        this.name = name;
        this.points = points;
        this.startStandId = startStandId;
        this.finishStandId = finishStandId;
    }
}

File: ra-front/src/app/model/MapAreas/MovementPaths/MovementPath.ts

import {UniversalPoint} from "../UniversalPoint";

export class MovementPath {
  id: string;
  name: string;
  points:UniversalPoint[];
  startStandId:string;
  finishStandId:string;


  constructor(id: string, name: string, points: UniversalPoint[], startStandId: string, finishStandId: string) {
    this.id = id;
    this.name = name;
    this.points = points;
    this.startStandId = startStandId;
    this.finishStandId = finishStandId;
  }
}
